### PR TITLE
fix hub and tag override for operator test

### PR DESCRIPTION
**Pull request by richardwxn**
1. fix hub and tag override
2. delete installed-state CR that might be in the cluster created before by previous manifest apply run

### CODE REVIEW

File: tests/integration/operator/main_test.go [hunk 1]
```diff
@@ -24,7 +24,6 @@ import (
 func TestMain(m *testing.M) {
 	framework.
 		NewSuite("operator_controller", m).
-		Skip("https://github.com/istio/istio/issues/22969").
```
Comment (howardjohn) [inline tests/integration/operator/main_test.go:27]: Can we tweak the makefile (temporarily, not to merge) to run what we run in postsubmit here before we merge this again?

**Review by howardjohn — CHANGES_REQUESTED** [review]

**Comment by dex80526** [comment]
@richardwxn: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: operator/pkg/helmreconciler/reconciler.go [hunk 2]
```diff
@@ -188,14 +197,86 @@ func (h *HelmReconciler) processRecursive(manifests name.ManifestMap) *v1alpha1.
 	}
 	wg.Wait()
 
+	ostatus := overallStatus(componentStatus)
+	if ostatus == v1alpha1.InstallStatus_HEALTHY {
+		if err := h.checkResourceStatus(&componentStatus, &ostatus); err != nil {
```
Comment (howardjohn) [inline operator/pkg/helmreconciler/reconciler.go:202]: Why do we need to do another wait here, when in `ApplyManifest` we can wait already?

**Review by howardjohn — COMMENTED** [review]
LGTM, not sure we need the extra wait logic though? But I am less familiar than @ostromart . Can you fix the lint error and we can followup with that once we can get the test enabled?

File: operator/pkg/helmreconciler/reconciler.go (hunk 2 above)
Comment (richardwxn) [reply operator/pkg/helmreconciler/reconciler.go:202]: that's for the case when user just update with running kubectl apply -f newcr

Comment (richardwxn) [reply operator/pkg/helmreconciler/reconciler.go:202]: ah actually now the new code would handles that case using the same flow though.. let me double check with Martin

Comment (ostromart) [reply operator/pkg/helmreconciler/reconciler.go:202]: correct, this code is not required.

Comment (richardwxn) [reply operator/pkg/helmreconciler/reconciler.go]: removed

**Comment by richardwxn** [comment]
result is here: https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/23099/integ-operator-controller-tests_istio/1225

**Comment by richardwxn** [comment]
can you PTAL again? @howardjohn @ostromart

**Review by howardjohn — APPROVED** [review]

**Pull request merged**